metaflow/plugins/kfp/kfp.py

```diff
@@ -705,21 +720,26 @@ def pipeline_transform(op: ContainerOp):
                             ),
                         )
                     )
+                # adding in additional env variable for spark to identify if workflow was
+                # launched from a notebook in an individual namespace.
+                env_vars = {
+                    "INDIVIDUAL_NAMESPACE": INDIVIDUAL_NAMESPACE,
+                }
                 # add in env variable for ServiceAccount for Zillow Spark solution
                 if KUBERNETES_SERVICE_ACCOUNT:
-                    env_var = {
-                        "METAFLOW_KUBERNETES_SERVICE_ACCOUNT": KUBERNETES_SERVICE_ACCOUNT
-                    }
-                    # need to be added separately from above as there is no valueFrom/fieldRef from the env
-                    # var. leaving as a list format in the event future env variables need to be added without
-                    # a fieldRef value_from similar to this env variable.
-                    for name, resource in env_var.items():
-                        op.container.add_env_variable(
-                            V1EnvVar(
-                                name=name,
-                                value=resource,
-                            )
+                    env_vars[
+                        "METAFLOW_KUBERNETES_SERVICE_ACCOUNT"
+                    ] = KUBERNETES_SERVICE_ACCOUNT
+                # need to be added separately from above as there is no valueFrom/fieldRef from the env
+                # var. leaving as a list format in the event future env variables need to be added without
+                # a fieldRef value_from similar to this env variable.
+                for name, resource in env_vars.items():
+                    op.container.add_env_variable(
+                        V1EnvVar(
+                            name=name,
+                            value=resource,
                         )
+                    )
```

Comment:
Looks like op.container.add_env_variable is already called on line 715 for part of the env_vars list. Will line 736 to 742 double add some env vars?  Maybe line 714 to 722 can be removed.

Reply:
it does not. the var env_vars gets reset at line 725 cause the variables we add in lines 736-742 add those env variables to the container with `name: value`. Lines 714-722 add the env vars set from lines 705-713 with the env var format `name: value_from` which is a different env var format. Both are needed and will not double add env variables.